x/config/doc.go

```diff
@@ -0,0 +1,186 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package config implements a generic configuration system that may be used
+// to build YARPC Dispatchers from configurations specified in different
+// markup formats.
+//
+// Usage
+//
+// To build a Dispatcher, set up a Loader and inform it about the different
+// transports that it needs to support. Use Load or LoadYAML to load a Builder
+// from the given configuration.
+//
+// 	loader := config.NewLoader()
+// 	http.RegisterTransport(loader)
+// 	redis.RegisterTransport(loader)
+//
+// 	builder, err := loader.LoadYAML(yamlConfig)
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+//
+// 	dispatcher, err := builder.BuildDispatcher()
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+//
+// Configuration parameters for the different transports, inbounds, and
+// outbounds are defined in the types fed into the TransportSpec that was
+// registered against the Loader.
+//
+// Configuration
+//
+// The configuration may be specified in YAML or as any Go-level
+// map[string]interface{}. The examples below use YAML for illustration
+// purposes.
+//
+// The configuration accepts the following top-level attributes: name,
+// transports, inbounds, and outbounds.
+//
+// 	name: myservice
+// 	inbounds:
+// 	  # ...
+// 	outbounds:
+// 	  # ...
+// 	transports:
+// 	  # ...
+//
+// Where name specifies the name of the current service. See the following
+// sections for details on the transports, inbounds, and outbounds keys in the
+// configuration.
+//
+// Inbound Configuration
+//
+// The 'inbounds' attribute configures the different ways in which the service
+// receives requests. It is represented as a mapping between inbound transport
+// type and its configuration. For example, the following states that we want
+// to receive requests over HTTP and Redis.
+//
+// 	inbounds:
+// 	  redis:
+// 	    # ...
+// 	  http:
+// 	    # ...
+//
+// (For details on the configuration parameters of individual transport types,
+// check the documentation for the corresponding transport package.)
+//
+// If you want multiple inbounds of the same type, specify a different name
+// for it and add a 'type' attribute to its configuration:
+//
+// 	inbounds:
+// 	  http:
+// 	    # ...
+// 	  http-deprecated:
+// 	    type: http
+// 	    # ...
+//
+// Any inbound can be disabled by adding a 'disabled' attribute.
+//
+// 	inbounds:
+// 	  http:
+// 	    # ...
+// 	  http-deprecated:
+// 	    type: http
+// 	    disabled: true
```

Comment:
Since inbounds are represented as a map instead of a list, you need a way to delete an inbound in an extending config file. For example, disabling gRPC in production while it is just being experimented in development.